### PR TITLE
feat(wallet): token v4

### PR DESCRIPTION
**Pull request by thesimplekid**
supersedes #84

### CODE REVIEW

**Comment by davidcaseria** [comment]
Are V4 tokens still going to support multiple mints?

**Comment by thesimplekid** [comment]
> Are V4 tokens still going to support multiple mints?

Not sure, I would prefer they don't. I left a comment on the nuts [PR](https://github.com/cashubtc/nuts/pull/109#discussion_r1629740064) . This is just you're work from the other PR I copied onto main since there are been a lot of changes merging was a mess. Hope that 's okay. Still gonna wait on the spec to merge this, should be an easy enough change if the multi mints are removed.

**Comment by davidcaseria** [comment]
Sounds good. I was holding off because I knew there was an open discussion and didn't want to keep having to make changes.

I also prefer if multi-mints were no longer supported.

File: crates/cdk/src/nuts/nut00/token.rs [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+            _ => return Err(Error::UnsupportedToken),
+        };
+
+        let decode_config = general_purpose::GeneralPurposeConfig::new()
+            .with_decode_padding_mode(base64::engine::DecodePaddingMode::Indifferent);
+        let decoded = GeneralPurpose::new(&alphabet::URL_SAFE, decode_config).decode(s)?;
+        let token: TokenV4 = ciborium::from_reader(&decoded[..])?;
+        Ok(token)
+    }
+}
```
Comment (thesimplekid) [inline crates/cdk/src/nuts/nut00/token.rs:337]: Remove v3

**Review by davidcaseria — APPROVED** [review]

**Pull request merged**